### PR TITLE
Policy evaluations frontend

### PR DESCRIPTION
**Pull request by salazarm**
## Summary & Motivation

Adds the new Automation page based off the new figmas,
Also adds a feature gate to allow reverting back the old page

## How I Tested These Changes
Tested:

- Viewing unpartitioned asset history
- Viewing partitioned asset history and selection a partition to view
- Viewing multi-partitioned asset history and selection a partition to view

Tested turning on the feature gate and that the old experience still worked:
Screenshot 2024-01-10 at 5 16 38 PM
Screenshot 2024-01-10 at 5 26 00 PM
Screenshot 2024-01-10 at 5 24 58 PM
Screenshot 2024-01-10 at 5 20 19 PM
Screenshot 2024-01-10 at 5 20 12 PM

Screenshot 2024-01-10 at 5 17 32 PM

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-components/src/components/Suggest.tsx [hunk 1]
```diff
@@ -99,23 +99,25 @@ export const Suggest = <T,>(props: Props<T>) => {
...
-            rowHeight={itemHeight}
-            rowRenderer={(a: any) => (
-              <div key={a.index} style={a.style}>
-                {props.renderItem(props.filteredItems[a.index] as T, a.index)}
-              </div>
-            )}
-            width={menuWidth}
-            height={Math.min(props.filteredItems.length * itemHeight, itemHeight * VISIBLE_ITEMS)}
-          />
+          <div style={{overscrollBehavior: 'contain'}}>
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-components/src/components/Suggest.tsx:102]: I don't know if I realized that this was using `react-virtualized` instead of tanstack, we should probably change that...

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeLeftPanel.tsx [hunk 2]
```diff
@@ -50,48 +60,82 @@ export const AutomaterializeLeftPanel = ({
...
+          {evaluations.map((evaluation) => {
+            const isSelected = selectedEvaluation?.id === evaluation.id;
+
+            const hasRequested = evaluation.numRequested > 0;
+
+            function status() {
+              if (hasRequested) {
+                if (definition?.partitionDefinition) {
+                  return (
+                    <Caption>{numberFormatter.format(evaluation.numRequested)} Requested</Caption>
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeLeftPanel.tsx:103]: Lowercase `requested`

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeLeftPanel.tsx [hunk 3]
```diff
@@ -50,48 +60,82 @@ export const AutomaterializeLeftPanel = ({
...
+      <Box padding={{vertical: 12, horizontal: 24}} border="bottom">
+        <Subtitle1>Evaluations</Subtitle1>
+      </Box>
+      <Box
+        padding={{bottom: 8, horizontal: 12}}
+        style={{flex: 1, minHeight: 0, overflowY: 'auto'}}
+        flex={{grow: 1, direction: 'column'}}
+      >
+        <Box border="bottom" padding={{top: 8, bottom: 12, left: 12, right: 8}}>
+          <Link to="/overview/automaterialze">
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeLeftPanel.tsx:83]: This doesn't apply to the sensor case though, right?

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeLeftPanel.tsx [hunk 4]
```diff
@@ -50,48 +60,82 @@ export const AutomaterializeLeftPanel = ({
...
+            function status() {
+              if (hasRequested) {
+                if (definition?.partitionDefinition) {
+                  return (
+                    <Caption>{numberFormatter.format(evaluation.numRequested)} Requested</Caption>
+                  );
+                }
+                return <Caption>Requested</Caption>;
+              }
+              return <Caption>Not Requested</Caption>;
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeLeftPanel.tsx:108]: Same

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeLeftPanel.tsx [hunk 5]
```diff
@@ -50,48 +60,82 @@ export const AutomaterializeLeftPanel = ({
...
+                    <StatusDot
+                      $color={
+                        evaluation.numRequested ? colorAccentGreen() : colorBackgroundDisabled()
+                      }
+                    />
+                    <span style={evaluation.numRequested ? {color: colorTextGreen()} : undefined}>
+                      <TimestampDisplay timestamp={evaluation.timestamp} />
+                    </span>
+                  </Box>
+                  <div style={{paddingLeft: 22}}>{status()}</div>
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeLeftPanel.tsx:130]: Bit of an unusual padding value

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeRunsTable.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  if (!runIds.length) {
+    return (
+      <Body2 color={colorTextLighter()} style={{paddingBottom: 32}}>
+        None
+      </Body2>
+    );
+  }
+
+  if (error) {
+    return <Body2>An error occurred fetching runs. Check your network status</Body2>;
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeRunsTable.tsx:38]: Seems more likely to be a GraphQL error than a network error here, right?

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeRunsTable.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    },
+    skip: !runIds.length,
+  });
+
+  if (!runIds.length) {
+    return (
+      <Body2 color={colorTextLighter()} style={{paddingBottom: 32}}>
+        None
+      </Body2>
+    );
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeRunsTable.tsx:34]: Is there a `NonIdealState` we should use here?

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PartitionSegmentWithPopover.tsx [hunk 8]
```diff
@@ -34,45 +38,72 @@ const statusToColors = (status: AssetConditionEvaluationStatus) => {
...
+        <PartitionSubsetList
+          description={description}
+          status={status}
+          subset={subset}
+          selectPartition={selectPartition}
+        />
+      }
+    >
+      <Tag intent={intent}>
+        {numberFormatter.format(count)} {status.charAt(0) + status.toLowerCase().slice(1)}
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PartitionSegmentWithPopover.tsx:90]: Not really a fan of slicing strings for copy like this. Maybe put it in a span and apply `text-transform`?

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PolicyEvaluationTable.tsx [hunk 9]
```diff
@@ -163,29 +217,29 @@ const PartitionedPolicyEvaluationTable = ({
...
                       subset={candidateSubset}
-                      width={Math.ceil((numSkipped / total) * FULL_SEGMENTS_WIDTH)}
+                      selectPartition={selectPartition}
                     />
                   ) : null}
                 </Box>
               </td>
               <td>
-                <TimeElapsed startUnix={startTimestamp} endUnix={endTimestamp} />
+                <TimeElapsed startUnix={startTimestamp} endUnix={endTimestamp} msec />
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PolicyEvaluationTable.tsx:242]: Looks like this was merged as `showMsec`?

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__fixtures__/AutoMaterializePolicyPage.fixtures.ts [hunk 10]
```diff
@@ -285,9 +251,7 @@ export const Evaluations = {
           }),
           currentAutoMaterializeEvaluationId: 1000,
         }),
-        autoMaterializeAssetEvaluationsOrError: buildAutoMaterializeAssetEvaluationRecords({
-          records: [],
-        }),
+        assetConditionEvaluationRecordsOrError: buildAssetConditionEvaluationRecords() as any,
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__fixtures__/AutoMaterializePolicyPage.fixtures.ts:254]: Is it necessary to `any` these? It implies there's a problem somewhere if these had to be any'd...

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__stories__/AutoMaterializePolicyPage.stories.tsx [hunk 11]
```diff
@@ -16,24 +16,16 @@ const path = ['test'];
 
 export const EmptyState = () => {
   return (
-    <MockedProvider
-      mocks={[Policies.NoAutomaterializeNoFreshnessPolicy(path), Evaluations.None(path, true)]}
-    >
-      <AssetAutomaterializePolicyPage assetKey={{path}} assetHasDefinedPartitions />
+    <MockedProvider mocks={[Evaluations.None(path, true)]}>
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__stories__/AutoMaterializePolicyPage.stories.tsx:19]: Can't really tell what `true` means here.

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__stories__/AutomaterializeLeftList.stories.tsx [hunk 12]
```diff
@@ -36,17 +35,16 @@ export const WithPartitions = () => {
...
 
   const evaluations = buildEvaluationRecordsWithoutPartitions();
 
   return (
     <div style={{width: '320px'}}>
       <AutomaterializeLeftList
-        assetHasDefinedPartitions={false}
-        evaluations={evaluations}
-        onSelectEvaluation={(evaluation: AutoMaterializeEvaluationRecordItemFragment) =>
+        evaluations={evaluations as any}
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__stories__/AutomaterializeLeftList.stories.tsx:46]: `any` necessary?

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__stories__/AutomaterializeMiddlePanel.stories.tsx [hunk 13]
```diff
@@ -19,32 +25,41 @@ const path = ['test'];
...
           assetKey={{path}}
-          assetHasDefinedPartitions={false}
+          definition={
+            buildAssetNode({
+              partitionKeysByDimension: [
+                buildDimensionPartitionKeys({
+                  partitionKeys: ['testing', 'testing2'],
+                }),
+              ],
+            }) as any
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__stories__/AutomaterializeMiddlePanel.stories.tsx:49]: ?

I know these are `stories.tsx` and `fixtures.tsx` but it makes me concerned that there are deeper type issues.

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/flattenEvaluations.tsx [hunk 14]
```diff
@@ -1,42 +1,64 @@
...
+  | SpecificPartitionAssetConditionEvaluationNodeFragment;
+
+type FlattenedEvaluation =
+  | FlattenedConditionEvaluation<PartitionedAssetConditionEvaluationNodeFragment>
+  | FlattenedConditionEvaluation<UnpartitionedAssetConditionEvaluationNodeFragment>
+  | FlattenedConditionEvaluation<SpecificPartitionAssetConditionEvaluationNodeFragment>;
+
+export const flattenEvaluations = (
+  evaluationRecord: Pick<AssetConditionEvaluationRecordFragment, 'evaluation'>,
+) => {
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/flattenEvaluations.tsx:29]: Why the change away from the generic type?

File: js_modules/dagster-ui/packages/ui-components/src/components/Suggest.tsx (hunk 1 above)
Comment (hellendag) [reply js_modules/dagster-ui/packages/ui-components/src/components/Suggest.tsx:102]: (Not in this PR of course)

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeLeftPanel.tsx (hunk 3 above)
Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeLeftPanel.tsx:83]: ah yeah

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PolicyEvaluationTable.tsx (hunk 9 above)
Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PolicyEvaluationTable.tsx:242]: yeah I need to rebase

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__fixtures__/AutoMaterializePolicyPage.fixtures.ts (hunk 10 above)
Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__fixtures__/AutoMaterializePolicyPage.fixtures.ts:254]: yeah the problem is that we our type generation doesn't work with polymorphic interfaces and MetadataEntry is polymorphic

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__stories__/AutomaterializeLeftList.stories.tsx (hunk 12 above)
Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__stories__/AutomaterializeLeftList.stories.tsx:46]: yeah see aboe

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__stories__/AutomaterializeMiddlePanel.stories.tsx (hunk 13 above)
Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__stories__/AutomaterializeMiddlePanel.stories.tsx:49]: same as above

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/flattenEvaluations.tsx (hunk 14 above)
Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/flattenEvaluations.tsx:29]: It doesn't work unfortunately because of how the backend is typed

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeRunsTable.tsx (hunk 6 above)
Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeRunsTable.tsx:38]: I'm not sure tbh, I've seen this return an error when it's actually a network error though

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PartitionSegmentWithPopover.tsx (hunk 8 above)
Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PartitionSegmentWithPopover.tsx:79]: does text-transform have an option to only capitalize the first letter?

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__stories__/AutoMaterializePolicyPage.stories.tsx (hunk 11 above)
Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__stories__/AutoMaterializePolicyPage.stories.tsx:19]: Screenshot 2024-01-17 at 4 13 43 PM
Screenshot 2024-01-17 at 4 13 33 PM

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PartitionSegmentWithPopover.tsx (hunk 8 above)
Comment (hellendag) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PartitionSegmentWithPopover.tsx:79]: `text-transform: capitalize` seems correct. [[link](https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform#example_using_capitalize_general)]

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PolicyEvaluationTable.tsx [hunk 15]
```diff
@@ -144,48 +226,47 @@ const PartitionedPolicyEvaluationTable = ({
...
+                  isCollapsed={uncollapsedRecords.has(evaluation.uniqueId)}
+                  hasChildren={evaluation.childUniqueIds.length > 0}
                 />
               </td>
+              <td>
+                {consideredPartitions ? (
+                  consideredPartitions
+                ) : (
+                  <Box flex={{direction: 'row', gap: 4, alignItems: 'center'}}>
+                    All{' '}
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PolicyEvaluationTable.tsx:248]: Put this in a `div` without the `' '` to let the flex gap do the spacing?

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PartitionSegmentWithPopover.tsx (hunk 8 above)
Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PartitionSegmentWithPopover.tsx:79]: It left "LONDON" complete capitalized so it's not correct.

File: js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__stories__/AutoMaterializePolicyPage.stories.tsx (hunk 11 above)
Comment (hellendag) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/__stories__/AutoMaterializePolicyPage.stories.tsx:19]: Sorry, my point was that when reading the function name and the args (without being able to click through to the definition), it's unclear what `true` vs. `false` actually does. Clarifying the value being passed in by using a config object for "named" vars, using an enum value, or perhaps creating a separate `NoneSingle` method, would make this clearer. I know it's just a fixture method though, so not a huge deal. :)

**Review by hellendag — APPROVED** [review]
Would love if the `any`s can be cleaned up in a followup by building a concrete value for the problematic interface. Even if the `any` values are just living in stories/tests/fixtures, it makes me itchy to have all type safety removed from those values in perpetuity.
